README.md

```diff
@@ -52,9 +54,25 @@ You can also pass the name of your project to the tool using the `--name` or `-n
 cargo generate --git https://github.com/githubusername/mytemplate.git --name myproject
 ```
 
-## Favorites
+### Template in a local directory, or git sub-directory
+
+Besides `--git` to specify the template repository, `cargo-generate` supports a `--dir` option to specify a local directory as the template.
+
+```sh
+cargo generate --dir ~/my-template-directory --name myproject
+```
+
+Using `--dir` and `--git` together, allows `cargo-generate` to use a sub-directory inside the git repository as the template.
```

Comment:
This is somehow really confusing to me. When I omit `--git` and have only `--dir` it is all good and clear. Makes total sense. But when I use `--git` and `--dir` I'm super puzzled. Nothing makes sense anymore in my head. 

I get the idea, and we've discussed the behaviour of `--pick` on #291, so that in general I'd say let's implement that behaviour there, and let's not reuse `--dir` for this 2nd different use case.

My reasoning is: if I'm confused by the duality of `--dir` when used in combination, than there is at least one person out there getting confused too.

Reply:
Your reasoning here is sound - and the reason I got confused in #291 - Thus I agree.

Got to admit though, the functionality was right there!!! So instead of simply disabling it I added the `--pick` to clear up the usage pattern. Call it a 2 for 1.